skfda/representation/basis.py

```diff
@@ -1444,6 +1444,31 @@ class FDataBasis(FData):
             ...)
 
     """
+    class _CoordinateIterator:
+        """Internal class to iterate through the image coordinates.
+
+        Dummy object. Should be change to support multidimensional objects.
+
+        """
+        def __init__(self, fdatabasis):
+            """Create an iterator through the image coordinates."""
+            self._fdatabasis = fdatabasis
```

Comment:
Maybe this should be a weak reference, to not create cycles? Another option is that the FData does not store the iterator, generating another one as needed.

Reply:
Solved in 10565de, now the call to `coordinates` creates a new iterator. The FData does not 
store the iterator.